cowprotocol/cowswap
# fix: remove uknown price impact warning flickering

# Summary

Fixes: #1515

<img width="515" alt="image" src="https://github.com/cowprotocol/cowswap/assets/7122625/d8423a90-a56b-4f54-b08f-161b41bed8f4">

First of all, why the issue is happening?
Because price impact state is based on trade USD amounts (`useTradeUsdAmounts`).
The hooks starts loading USD amounts only when trade amounts are set in turn this mean we need a quote to be fetched to start calculating the price impact. This race condition creates a time when we don't know the price impact and don't load it, so we consider this case as "Price impact is unknown".

To fix the issue, I've added two arguments for `useTradeUsdAmounts()` hook. Now it start updating USD amounts state once both trade assets are set. It means, we don't wait for a quote.

  # To Test

It's very easy to reproduce the issue on Goerli.

For each swap/llimit/twap:

1. Select WETH -> COW
2. Wait until getting quote
- [ ] If fiat prices fetching has taken less than 3 seconds, then the price impact warning should not be displayed
3. Change one of assets
- [ ] Should work the same as in the previous point
4. Select WETH -> DAI
5. Wait until getting quote
- [ ] since there are no fiat prices for DAI on Goerli, the warning should be displayed in 3 seconds
